### PR TITLE
KEYCLOAK-5352 Basic Auth fails if password contains a ':'

### PR DESCRIPTION
**Pull request by ricehull**
Modified 3 source files:
./adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/BasicAuthRequestAuthenticator.java
./core/src/main/java/org/keycloak/util/BasicAuthHelper.java
./services/src/main/java/org/keycloak/protocol/saml/profile/ecp/authenticator/HttpBasicAuthenticator.java

### CODE REVIEW

File: adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/BasicAuthRequestAuthenticator.java [hunk 1]
```diff
@@ -71,9 +71,12 @@ public AuthOutcome authenticate(HttpFacade exchange)  {
         AccessTokenResponse atr=null;        
         try {
             String userpw=new String(Base64.decode(tokenString));
-            String[] parts=userpw.split(":");
-            
-            atr = getToken(parts[0], parts[1]);
+            int seperatorIndex = userpw.indexOf(":");
+            String user = userpw.substring(0, seperatorIndex);
+            String pw = userpw.substring(seperatorIndex + 1);
+            log.debug("user: " + user);
```
Comment (stianst) [inline adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/BasicAuthRequestAuthenticator.java]: This is not good logging. Please use a sensible message, use a single log entry and make it trace

File: services/src/main/java/org/keycloak/protocol/saml/profile/ecp/authenticator/HttpBasicAuthenticator.java [hunk 2]
```diff
@@ -94,7 +97,14 @@ protected void notValidCredentialsAction(final AuthenticationFlowContext context
         }
 
         try {
-            return new String(Base64.decode(credentials)).split(":");
+            String val = new String(Base64.decode(credentials));
+            int seperatorIndex = val.indexOf(":");
+            if(seperatorIndex == -1) return null;
+            String user = val.substring(0, seperatorIndex);
+            String pw = val.substring(seperatorIndex + 1);
+            log.debug("user: " + user);
```
Comment (stianst) [inline services/src/main/java/org/keycloak/protocol/saml/profile/ecp/authenticator/HttpBasicAuthenticator.java]: See earlier comment on logging

**Review by stianst — CHANGES_REQUESTED** [review]

**Review by ricehull — COMMENTED** [review]
Done for the log.

**Comment by stianst** [comment]
Thinking about it some more we shouldn't log user passwords. It would probably be best to just remove the logging.

**Comment by ricehull** [comment]
I thought this is a debug log, so I logged the user name and password, But I think your comments is more resonable, we can remove the log of the modification.

**Comment by ricehull** [comment]
It's failed as yesterday, looks nothing to do with my change, I just remove the logs for the methods.

**Comment by stianst** [comment]
The failing test is one we've had problems with so it's not your fault. Thanks for the fix

**Comment by stianst** [comment]
Looks like some tests are failing due to this.. Specifically UndertowDemoFilterServletAdapterTest.testBasicAuth which isn't ran on Travis for some reason.

**Comment by stianst** [comment]
Never mind. I've tried to run the affected test with the last commit before yours and it's broken there as well. So just happen to be broken at the same time.

**Pull request merged**